compiler/rustc_const_eval/src/const_eval/valtrees.rs

```diff
@@ -290,7 +290,7 @@ pub fn valtree_to_const_value<'tcx>(
             debug!(?place);
 
             valtree_into_mplace(&mut ecx, &mut place, valtree);
-            dump_place(&ecx, place.into());
+            dump_place(&ecx, place.clone().into());
```

Comment:
This seems rather silly: we have a whole bunch of `clone().into()` just for all this dumping. Can't we just inline the `trace!("{:?}", ecx.dump_place(*place))` everywhere? Then at least the clone/into is avoided when `trace!` is disabled...

Reply:
actually all the callers to `dump_place` pass a `MplaceTy` and `.into()` it. So we may as well do that `.into()` within the function 😆